src/main/java/core/dao/ManufacturerDaoImpl.java

```diff
@@ -0,0 +1,42 @@
+package core.dao;
+
+import core.db.Storage;
+import core.lib.Dao;
+import core.model.Manufacturer;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+@Dao
+public class ManufacturerDaoImpl implements ManufacturerDao {
+    @Override
+    public Manufacturer create(Manufacturer manufacturer) {
+        Storage.addManufacturer(manufacturer);
+        return manufacturer;
+    }
+
+    @Override
+    public Optional<Manufacturer> get(Long id) {
+        return Storage.manufacturers.stream()
+                .filter(x -> Objects.equals(x.getId(),id))
+                .findFirst();
+    }
+
+    @Override
+    public List<Manufacturer> getAll() {
+        return Storage.manufacturers;
+    }
+
+    @Override
+    public Manufacturer update(Manufacturer manufacturer) {
+        Manufacturer old = get(manufacturer.getId()).get();
+        old.setName(manufacturer.getName());
+        old.setCountry(manufacturer.getCountry());
+        return old;
```

Comment:
this code isn't flexible. We should change the logic of this method if the manufacturer object is changed
let's use the method `set` from the list for replacing an object